src/python/pants/engine/internals/native_engine.pyi

```diff
@@ -24,7 +24,9 @@ from pants.engine.process import InteractiveProcessResult
 class AddressParseException(Exception):
     pass
 
-def address_parse(spec: str) -> tuple[str, str | None, str | None, tuple[tuple[str, str], ...]]: ...
+def address_spec_parse(
+    spec: str,
+) -> tuple[bool, tuple[str, str | None, str | None, tuple[tuple[str, str], ...]], str | None]: ...
```

Comment:
Not worth a CI round, but it would be good to document the semantics of the return value, since it's a more complex structure now.

Reply:
Yea, this has almost met the bar for a dedicated class at this point. I tried to use type aliases in here, but failed as well. I think that the documentation on the rust side does cover the base to a degree though.

Reply:
+1 to class